python/tests/test_regression.py

```diff
@@ -0,0 +1,62 @@
+from vw_executor.vw import Vw
+from vw_executor.vw_opts import Grid
+from numpy.testing import assert_allclose
+import pandas as pd
+import numpy as np
+import random
+
+with open("regression1.txt", "w") as f:
+    for i in range(10000):
+        x = random.uniform(1, 100)
+        y = 5
+        f.write(f"{y} |f x:{x}\n")
+
+
+def core_test(files, grid, outputs, job_assert):
+    vw = Vw(".vw_cache", reset=True)
+    result = vw.train(files, grid, outputs)
+    job_assert(result)
+
+
+def test_regression_weight():
+    options = Grid(
+        {
+            "#base": ["-P 50000 --preserve_performance_counters --save_resume "],
+            "#reg": ["", "--coin", "--ftrl", "--pistol"],
+        }
+    )
+
+    def assert_job(job):
+        tolerance = 1
+        constant = 5
+        for j in job:
+            assert j.status.value == 3
+            data = j.outputs["--readable_model"]
+            with open(data[0], "r") as f:
+                data = f.readlines()
+            data = [i.strip() for i in data]
+            weight_b = float(data[-1].split(":")[1].split(" ")[0])
```

Comment:
https://github.com/VowpalWabbit/data-science/blob/b9cadfed0622e99050afa5685d77c33aee8703c5/from_mwt_ds/DataScience/vw_executor/artifacts.py#L220

you can have access to weights as job[0].Model9('--readable_model').weights